### ISSUE TITLE
Restored session can expand/collapse wrong trees

### ISSUE THREAD

**Issue by lbmaian** [piroor/treestyletab#1447]
## Short description
Restored session can expand/collapse wrong trees. I believe this is a regression since 2.02, since I can only reproduce on the latest nightly, and not in 2.02 itself.

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Tree structure:
```
A
\- B (collapsed)
C (selected)
\- D
```
 4. Select tab A:
```
A (selected)
\- B
C
\- D (collapsed)
```
 5. Restart Firefox

## Expected result
```
A (selected)
\- B
C
\- D (collapsed)
```

## Actual result
```
A (selected)
\- B (collapsed)
C
\- D
```

## Environment

 * Platform (OS): Windows 10 Pro, Version 1709, Build 16299.15
 * Version of Firefox: Developer Edition 57.0b6
 * Version (or revision) of Tree Style Tab: Latest nightly as of commit 1562e17661fb3947128a7c9722f2f8b19a6d3d33

**Comment by piroor** [comment]
I couldn't reproduce this anymore with latest development build of TST. Recent commits possibly affect to this issue. But the problem is similar to #1445, so those changes possibly don't affect. Could you try with the latest development build?

**Comment by lbmaian** [comment]
This still reproduces for me on the latest TST nightly (812348e94d4fdfc673047a53305c48f07b77bcc6), on both the latest Firefox nightly (58.0a1 (2017-10-09)) and the latest Firefox dev edition (57.0b6).

In fact, sometimes it triggers with #1445 if I close the browser fast enough.

**Comment by lbmaian** [comment]
Interestingly, if I close the browser window _right_ after selecting tab A, within a split second, the A/B tree does expand on restore, even if the bug in #1445 is almost always triggered.

**Comment by lbmaian** [comment]
This still occurs on the latest TST nightly for me (11657b6fc4fde29cbf80e8721e97313fa1954735).

**Comment by piroor** [comment]
Sorry I've successfully confirmed this.

**Comment by piroor** [comment]
After 507663c this problem doesn't happen for me. (Note that you need to wait for a while until Firefox's session store module flushes all data to the disk like #1445 .)

**Comment by lbmaian** [comment]
Okay, verified fixed on the just-released 2.03.

Confusingly, the latest nightly is actually outdated, at 2.0.2.5372, and apparently doesn't include that last fix (507663c578a41b8264144c8f6466b7416a8cf553).
